modules/local/staramr/main.nf

```diff
@@ -45,12 +45,17 @@ process STARAMR_SEARCH {
         -o ${prefix}_results \\
         $genome_filename
 
+
     # Add prefix ($meta.id) to the names of output files (allows for CSVTK module to concatenate files downstream)
     for f in ${prefix}_results/* ; do mv "\$f" \$(echo \$f | sed 's;/;/${prefix}_;'); done
 
     # Add extension (.staramr) to the names of output files (to simplify output differntiation amongst other modules)
     for f in ${prefix}_results/*{.tsv,.xlsx,.txt} ; do mv "\$f" \$(echo \$f | sed 's;\\.\\([^\\.]*\\)\$;\\.staramr\\.\\1;'); done
 
+    # Add a meta.irida column to the *_summary.staramr.tsv file to allow for meta.irida to be passed to the iridanext.output.json.gz
+    awk 'BEGIN{ FS = OFS = "\\t" } { print (NR==1? "IRIDA_ID" : "${meta.irida_id}"), \$0 }' ${prefix}_results/${prefix}_summary.staramr.tsv > tmp && mv tmp ${prefix}_results/${prefix}_summary.staramr.tsv
```

Comment:
Be really careful with code injection here.

Reply:
To avoid this we will limit the characters supported by the `meta.irida_id` to accept.
```
 // Remove characters from meta.irida_id that could cause issues in processes
 meta.irida_id = meta.irida_id.replaceAll(/[;\\#><|]/, '_')
```

Reply:
I still think this code is vulnerable due to the `ext.prefix`, they could easily supply a config file to modify it.
```
process {


    withName: STARAMR_SEARCH {

        ext.prefix = "test"
    }
}
```

Reply:
By, accident, not realizing the functionality of `ext.prefix` made the process unable to work with `ext.prefix` due to my `output block`
```
output:
    tuple val(meta), path("*_results/${meta.id}_results.staramr.xlsx")          , emit: results_xlsx
    tuple val(meta), path("*_results/${meta.id}_summary.staramr.tsv")           , emit: summary_tsv
    tuple val(meta), path("*_results/${meta.id}_detailed_summary.staramr.tsv")  , emit: detailed_summary_tsv
    tuple val(meta), path("*_results/${meta.id}_resfinder.staramr.tsv")         , emit: resfinder_tsv
    tuple val(meta), path("*_results/${meta.id}_plasmidfinder.staramr.tsv")     , emit: plasmidfinder_tsv
    tuple val(meta), path("*_results/${meta.id}_mlst.staramr.tsv")              , emit: mlst_tsv
    tuple val(meta), path("*_results/${meta.id}_settings.staramr.txt")          , emit: settings_txt
    tuple val(meta), path("*_results/${meta.id}_pointfinder.staramr.tsv")       , emit: pointfinder_tsv, optional: true
```
It always expects `meta.id = prefix`.

I don't know if we could axe the `ext.prefix` or come up with a safer way to modify the staramr tsv file.

Reply:
Another possible solution is to remove problematic characters in prefix before running the awk command

Reply:
Discussing a bit with @apetkau, we don't believe that code injection will be an issue for `meta.irida_id` due to the fact IRIDA-Next has guard rails in place, i.e. users cannot modify it. For `meta.id` we already check this (coming from code proposed in `speciesabundance` by @Kyla). Lastly, the only way to modify `prefix` is to pass a config file from the command line, which would be relevant for running locally.

I do not believe we need the change 1d209a8 and can consider this resolved. Unless there are still some objections.

Reply:
Reverted it here 1d829b3

Reply:
Thanks so much for the comment Eric, and investigation into this Steven.

The `meta.id` (which comes from the `sample_name`) is restricted in which characters it can contain via a regex at the start of the pipeline. The `meta.irida_id` comes from IRIDA Next, and so is not user-controlled (so there isn't risk of code injection when using with IRIDA Next).

In theory, we could add a more restrictive pattern for the `irida_id` in the JSON schemas. This might be something we could do to other pipelines as well. This adds another layer of protection against code injection (though already can't happen with IRIDA Next).

So, @sgsutcliffe I'm wondering if you could switch out (for the `sample` column in the JSON schema):

```
"pattern": "^\\S+$",
```

With:

```
"pattern": "^[A-Za-z0-9_.-]+$",
```

What do you think?

Reply:
My only thought against this would be that it would restrict people who are running it locally. But I will make the change, because I do not think it would cause any issues. If users want names that use other characters they can do that with the `sample_name`. `sample` really is just an IRIDA-Next column.

Reply:
Made the change b38c461